### DISCUSSION TITLE
Better Ticket reporting, new fields and data snapshots

### DISCUSSION THREAD

**Discussion by kdecks** [RetroAchievements/RAIntegration#760] [Ideas]
As suggested by **MutaWarrior** in the devs channel in Discord.

As a member of the admin team I want for the developers to have as easy a time as possible resolving tickets:

The enhancement idea is that when a ticket is submitted a snapshot of the current hitcounts for the achievement in question would be included as part of the ticket report. This could possibly be integrated into @meleu's recent memory inspector logic display update for tickets.

- The snapshot may vary in usefulness depending on how long after the ``didn't trigger`` type of ticket is reported.

I do not know how much this issue is a hybrid-web/integration issue. Regardless, anything that can help developers get a better clue to what is going wrong is for the best.

**Comment by Jamiras** [comment]
Related to https://github.com/RetroAchievements/RAWeb/issues/119.

As suggested, the data would be inaccurate for "didn't trigger" tickets, and may even be misleading. For example, a failed damageless boss achievement may be entirely reset when the player advances to the next stage, or a pre-condition may be reactivated after the boss is defeated.

And for "triggered at the wrong time" tickets, you should have a pretty good idea of the values as the achievement did trigger.

**Comment by julkasalsa** [comment]
I think such log could still be potentially very useful, especially for achievements, which require multiple checkpoints in a game with more chaotic and bigger memory, in a big system which could get affected in countless ways, with bigger chance for hard to find, random malfunction (RAP64). Something like this shouldn't be used as an evidence for the achievement being bugged, but in case of searching potential loopholes, when any piece of information like this could help you.

**Comment by kdecks** [comment]
## Better Ticket Reports - Another draft
> Response to feedback here posted by @Jamiras and discussions and thoughts I've had since. Shoutout to @JuliaWolska for helpful feedback.

### Mock-up appearance and "False Trigger" contextual questions
![image](https://user-images.githubusercontent.com/32706333/48612617-bcd72900-e946-11e8-8a71-7dd523f3cce1.png)

## Primary Changes to "Report Broken Achievements" (Front & Backend)
1. Rename "Report Broken Achievement" to "Report Achievement problem" to be more inclusive to new problem types. See below.
2. Add two new reportable problem types: 
   2a. ``Typo``
   2b. ``Description unclear``
3. Change trigger type titles for ``Triggered at Wrong Time`` and ``Didn't Trigger at all`` to something shorter. ``False Trigger`` / ``Trigger Absent``. (not important).
4. Selecting a problem type will add contextual questions for each type.
5. Selecting problem types ``False Trigger`` or ``Trigger Absent`` will add file upload options.
6. Collect new data snapshots from the game: (not shown to user)
   6a.  ``Snapshot of ticket's hits``.
   6b. ``Snapshot of current values for all conditions in the ticketed achievement``.
   6c. ``Snapshot of Rich Presence``.

- Use for ``Snapshot of ticket hits``: For ``False Trigger`` type problems is to see which ``alt group`` has triggered. Or for normal requirements with (0) hits asked, the current hit-count could be. (Both only slightly useful.) For ``Trigger Absent`` type problems is to see which conditions have and have not been met.
- Use for ``Snapshot of current values for all conditions in the ticketed achievement``: To help give context and current player position, compared to expected trigger point. To help get a clue about what might have gone wrong. To help verify usefulness of other snapshot data.
- Use for ``Snapshot of Rich Presence``: To give context and player position which will help verify accuracy of other snapshot data.

----
### "Trigger Absent": Contextual questions
What achievement requirements did you complete?
![image](https://user-images.githubusercontent.com/32706333/48614451-85b74680-e94b-11e8-9bd7-ccc670343bba.png)

What led up to earning this achievement?
   - What in game settings have you chosen? Paths you have taken?
   - How long have you been playing?
   - How many reattempts have you taken since the last game reset? Any other details?
![image](https://user-images.githubusercontent.com/32706333/48614427-76d09400-e94b-11e8-9445-bdd033c38897.png)
Are you on the same screen as when you reported it?
![image](https://user-images.githubusercontent.com/32706333/48614285-1e999200-e94b-11e8-882a-f146f82ae4e7.png)
How much time has passed in game since the Absent Trigger?
![image](https://user-images.githubusercontent.com/32706333/48614156-d5494280-e94a-11e8-9573-43b8123a4f80.png)

### "Typo in title or description": Contextual question
What is the typo, and what do you think it should say?
![image](https://user-images.githubusercontent.com/32706333/48614480-9667bc80-e94b-11e8-9c78-48172919d35a.png)

### "Description is unclear or confusing": Mock-up appearance and contextual questions
![image](https://user-images.githubusercontent.com/32706333/48615511-548c4580-e94e-11e8-9c29-c80117d37a78.png)


---
### Current reporting window for reference
![image](https://user-images.githubusercontent.com/32706333/48572317-5c080c00-e8c6-11e8-9068-2024d8999a8b.png)

----
#### Comments and disclaimers
> I'd like to highlight that I feel like above is a big undertaking, and I'm not sure how reasonable it is. But perhaps we can take some aspects of it into actual development.
>
> Also integrated ideas from https://github.com/RetroAchievements/RAWeb/issues/119, suggested by @magusLOGS
>
> One thing I don't feel 100% about is the questions... I'm going for descriptive clarity, so we can squeeze what we need out of the users.
>
> Another problem is too many questions lumped together. Maybe these can be split up.
>
> Want to disclaim that I'm not dead-set on any of these ideas. My goal is to identify problems and propose solutions, or to move the conversation towards those solutions.

**Comment by kdecks** [comment]
Another related idea is to integrate manual unlock requests into tickets for "Did not trigger at all" types.
